refactor(core): remove deprecated Renderer

crisbeto opened the pull request:
Removes the `Renderer` and related symbols which have been deprecated since version 4.

BREAKING CHANGES:
* `Renderer` has been removed. Use `Renderer2` instead.
* `RenderComponentType` has been removed. Use `RendererType2` instead.
* `RootRenderer` has been removed. Use `RendererFactory2` instead.

mary-poppins commented:
You can preview cc7a684 at https://pr33019-cc7a684.ngbuilds.io/.
You can preview 592eb23 at https://pr33019-592eb23.ngbuilds.io/.
You can preview 2573703 at https://pr33019-2573703.ngbuilds.io/.
You can preview 5eeb5cb at https://pr33019-5eeb5cb.ngbuilds.io/.

AndrewKushnir commented:
- [VE Presubmit](http://test/OCL:273204064:BASE:273204088:1570414399375:6d54bb27)
- [VE Global Presubmit](https://test.corp.google.com/ui#id=OCL:273204064:BASE:273204144:1570414445051:1e833f0b)

pkozlowski-opensource reviewed (approved):
Cool, LGTM! Hope it won't brake (too many) people and we can land this one!

AndrewKushnir commented:
FYI, VE presubmits (including a [global one](https://test.corp.google.com/ui#id=OCL:273204064:BASE:273304577:1570464326300:264c7fa0)) are successful. Thank you.

kara commented on packages/docs/web_workers/web_workers.md:
```diff
@@ -149,15 +149,15 @@ to the transformer until that bug is fixed.
-should use data bindings and if needed you can inject the `Renderer` along with your component's `ElementRef`
-directly into your component and use methods such as `setElementProperty`, `setElementAttribute`,
-`setElementClass`, `setElementStyle`, `invokeElementMethod`, and `setText`. Note that you **cannot** call
+should use data bindings and if needed you can inject the `Renderer2` along with your component's `ElementRef`
```
Same below

```suggestion
should use data bindings and if needed you can inject `Renderer2` along with your component's `ElementRef`
```

kara commented on aio/content/guide/deprecations.md:
```diff
@@ -96,9 +96,9 @@ Tip: In the [API reference section](api) of this doc site, deprecated APIs are i
-| [`RenderComponentType`](api/core/RenderComponentType) | [`RendererType2`](api/core/RendererType2) and  [`Renderer2`](api/core/Renderer2) | v4 | none |
-| [`Renderer`](api/core/Renderer) | [`Renderer2`](api/core/Renderer2) | v4 | none |
-| [`RootRenderer`](api/core/RootRenderer) | [`RendererFactory2`](api/core/RendererFactory2) | v4 | none |
+| `RenderComponentType` | [`RendererType2`](api/core/RendererType2) and  [`Renderer2`](api/core/Renderer2) | v4 | none |
```
I think we should remove these from the table in v9 or it looks like it's still deprecated. Users should still see it in v8 docs site.

kara reviewed (approved):
LGTM once comments are addressed

crisbeto commented:
The feedback has been addressed.

mary-poppins commented:
You can preview d581bb8 at https://pr33019-d581bb8.ngbuilds.io/.

kara commented:
note: verbal LGTM from igor for public api side

IgorMinar commented on aio/content/guide/deprecations.md:
```diff
@@ -40,8 +40,6 @@ v8 - v11
 | `@angular/core` | [`DefaultIterableDiffer`](#core) | <!--v7--> v9 |
 | `@angular/core` | [`ReflectiveKey`](#core) | <!--v8--> v9 |
 | `@angular/core` | [`RenderComponentType`](#core) | <!--v7--> v9 |
-| `@angular/core` | [`Renderer`](#core) | <!--v7--> v9 |
```
can you move this and the next hunk to the "removed api" section: https://angular.io/guide/deprecations#removed-apis thanks

crisbeto replied:
Done.

IgorMinar reviewed (approved):
one change requested for docs, but otherwise lgtm for API changes

crisbeto committed:
refactor(core): remove deprecated Renderer

Removes the `Renderer` and related symbols which have been deprecated since version 4.

BREAKING CHANGES:
* `Renderer` has been removed. Use `Renderer2` instead.
* `RenderComponentType` has been removed. Use `RendererType2` instead.
* `RootRenderer` has been removed. Use `RendererFactory2` instead.

mary-poppins commented:
You can preview a22e015 at https://pr33019-a22e015.ngbuilds.io/.

kara commented:
global approval
